controllers/argocd/role_test.go

```diff
@@ -151,6 +151,34 @@ func TestReconcileArgoCD_reconcileClusterRole(t *testing.T) {
 	assert.Contains(t, r.Client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleName}, reconciledClusterRole).Error(), "not found")
 }
 
+func TestReconcileArgoCD_reconcileClusterRole_disabled(t *testing.T) {
```

Comment:
It would also be good to test the case where:
- an `ArgoCD` is initially created with `a.Spec.DefaultClusterScopedRoleCreationDisabled = false`
- reconcile is called
- verify cluster roles are created
- then that `ArgoCD` is set `a.Spec.DefaultClusterScopedRoleCreationDisabled = true`
- verify cluster roles are deleted